tlaplus: 1.7.0 -> 1.7.1 base upon released jar file

florentc opened the pull request:
###### Motivation for this change

The current version is one year late compared to the latest stable release. In the previous tlaplus package, there was an issue with the tools when passing some command line arguments such as `-workers 4` for `tlc`. A java string method was not found and the program would not proceed correctly. 

###### Things done

This updates tlaplus to the most recent stable version and directly fetches the tlatools jar file from the official Github release page instead of building from the sources. This solves the command line arguments issue.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -3908,6 +3908,12 @@
     githubId = 183879;
     name = "Florian Klink";
   };
+  florentc = {
```
Please cherry-pick the commit into the PR where you are adding yourself as a maintainer to a package. This makes sure all maintainers entries are actually used and we do not collect stale ones.

florentc replied:
Would you like me to split the commit into two commits (one to add myself as a maintainer and one to update the package) or open another pull request to add myself as maintainer?

SuperSandro2000 replied:
That was the wrong template... You can do that in this PR.

That is the right one:

----

Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

florentc replied:
This is done.

SuperSandro2000 commented on pkgs/applications/science/logic/tlaplus/default.nix:
```diff
@@ -1,41 +1,37 @@
+  buildInputs = [ adoptopenjdk-bin ];
 
-  buildPhase = "ant -f tlatools/org.lamport.tlatools/customBuild.xml compile dist";
+  unpackPhase = ":";
```
```suggestion
  dontUnpack = true;
```

florentc committed:
maintainers: add florentc

florentc committed:
tlaplus: 1.7.0 -> 1.7.1

This updates tlaplus to the most recent stable version and directly
fetches the tlatools jar file from the official Github release page
instead of building from the sources. In the previous tlaplus nix
package, there was an issue with the tools when passing some command
line arguments such as `-workers 4` for TLC. A java string method was
not found and the program would not proceed correctly. This solves this
issue.
